lib/components/Editor/CustomExtensions/Image/ExtensionConfig.js

```diff
@@ -99,16 +99,15 @@ const ImageExtension = Image.extend({
   },
 });
 
-const upload = (file, url) =>
-  new Promise((resolve, reject) => {
-    const uploader = new DirectUpload(file, url);
-
-    uploader.create((error, { blob_url: imageUrl }) => {
-      if (error) return reject(error);
-
-      return resolve(imageUrl);
-    });
-  });
+const upload = async (file, url) => {
+  const uploader = new DirectUpload({ file, url });
+  try {
+    const { blob_url: imageUrl } = await uploader.create();
+    return Promise.resolve(imageUrl);
```

Comment:
Since this is already a async function, I think you can just return imageUrl directly which automatically becomes the resolved value of the promise